Lib/test/test_os.py

```diff
@@ -3328,40 +3238,52 @@ class TestSendfile(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        cls.key = threading_helper.threading_setup()
         create_file(os_helper.TESTFN, cls.DATA)
 
     @classmethod
     def tearDownClass(cls):
-        threading_helper.threading_cleanup(*cls.key)
         os_helper.unlink(os_helper.TESTFN)
 
-    def setUp(self):
-        self.server = SendfileTestServer((socket_helper.HOST, 0))
-        self.server.start()
+    @staticmethod
+    async def chunks(reader):
+        while not reader.at_eof():
+            yield await reader.read()
+
+    async def handle_new_client(self, reader, writer):
+        self.server_buffer = b''.join([x async for x in self.chunks(reader)])
+        writer.close()
+        self.server.close()  # The test server processes a single client only
+
+    async def asyncSetUp(self):
+        self.server_buffer = b''
+        cb = lambda reader, writer: self.handle_new_client(reader, writer)
+        self.server = await asyncio.start_server(cb, socket_helper.HOSTv4)
```

Comment:
It the callback is a function with `reader, writer` params, why not pass `self.handle_new_client` directly without the intervening lambda?

Reply:
Thank you, I didn't know that Python autocaptures `self` when obtaining a reference to a method so I've done it manually in the lambda first. I've thought that, like in C++ or JavaScript, just getting a method reference without calling it right there will require to specify a call target once more.

Reply:
It’s the beauty of descriptors!